eta/core/video.py

```diff
@@ -738,12 +738,19 @@ def get_frame_range(self):
         fns = self.get_frame_numbers()
         return (fns[0], fns[-1]) if fns else (None, None)
 
-    def merge_video_labels(self, video_labels):
+    def merge_video_labels(self, video_labels, reindex_objs=False):
         '''Merges the given VideoLabels into this labels.
 
         Args:
             video_labels: a VideoLabels instance
+            reindex_objs: whether to reindex objects in `self` before merging
+                so that all indices are different than indices in
+                `video_labels`
         '''
+        if reindex_objs:
+            self._reindex_objs(video_labels, "index")
+            self._reindex_objs(video_labels, "index_in_frame")
```

Comment:
`index_in_frame` means the position of the `DetectedObject` in the container in its associated frame. It's used to associated bounding boxes with, for example, features that are separately written to disk. So this isn't doing the right thing unfortunately

Reply:
Also, this implies that `index_in_frame` cannot be changed in a vacuum.

Or, rather, one really shouldn't be merging `VideoLabels` whose objects have `index_in_frame` if one actually cares about _using_ those values. I take the blame for this because the meaning of the field is not very clearly documented